src/test/java/school/hei/haapi/integration/PaginationIT.java

```diff
@@ -88,10 +105,48 @@ void student_pages_are_ordered_by_reference() throws ApiException {
     assertTrue(isBefore(page2.get(0).getRef(), page2.get(2).getRef()));
   }
 
+  @Test
+  void fees_page_are_ordered_by_due_datetime_desc() throws ApiException {
+    ApiClient student1Client = anApiClient(STUDENT1_TOKEN);
+    PayingApi api = new PayingApi(student1Client);
+    int pageSize = 2;
+
+    List<Fee> page1 = api.getStudentFees(STUDENT1_ID, 1, pageSize);
```

Comment:
Can you check whether we put ordered indexes (of the approriate orderding) on fee.due_datetime (DESC) and payment.creation_datetime (DESC or ASC, I don't remember)? https://www.postgresql.org/docs/current/indexes-ordering.html

When we now that we will ALWAYS (or very often) return a list ordered by some attribute, we add such ordered indexes. Warning: if the index order is DIFFERENT from the request submitted, then we may have dramatic result!

Reply:
Actually, we didn't index _yet_ the fee.due_datetime(DESC) and payment.creation_datetime(DESC). For our use case, the most recent (descending) order is the most efficient, and I don't think there is a case where we should change this order.

So I think we can put ordered (DESC) indexes for these fields.